### PR TITLE
[cc1352] fixes for integration with Texas Instruments examples

### PR DESCRIPTION
**Pull request by ltaoti**
Bugfixes for two issues discovered during OpenThread integration with Texas Instruments example applications:

1) Add const keyword to Dataset and DatasetLocal class constructor parameters in the header file, matching the constructor parameters in the source file. The Code Composer Studio linker throws an error if these constructor parameters don't match.

2) Fix a bug in the radio.c files for the CC1352, CC2652, and CC2650 platforms where the otPlatRadioEnable function would return OT_ERROR_BUSY even if there was no error. Fixed by correctly setting the return error code to OT_ERROR_NONE.

### CODE REVIEW

File: src/core/meshcop/dataset.hpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ class Dataset
      * @param[in]  aType       The type of the dataset, active or pending.
      *
      */
-    explicit Dataset(Tlv::Type aType);
```
Comment (jwhui) [inline src/core/meshcop/dataset.hpp]: I think we should not add `const` and remove `const` from the corresponding `*.cpp` files.

File: src/core/meshcop/dataset_local.hpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ class DatasetLocal : public InstanceLocator
      * @param[in]  aType      The type of the dataset, active or pending.
      *
      */
-    DatasetLocal(Instance &aInstance, Tlv::Type aType);
```
Comment (jwhui) [inline src/core/meshcop/dataset_local.hpp]: I think we should not add `const` and remove `const` from the corresponding `*.cpp` files.

**Comment by jwhui** [comment]
@ltaoti, thanks for this contribution! Just a minor comment about the `const`. Also looks like you need to get added to TI's CLA group - @srickardti should be able to help.

**Comment by srickardti** [comment]
@jwhui Yep, working on the CLA right now.

File: src/core/meshcop/dataset.hpp (hunk 1 above)
Comment (ltaoti) [reply src/core/meshcop/dataset.hpp]: Sure, sounds good.

**Pull request merged**